Source/CombatExtended/Harmony/Harmony_GenRadial.cs

```diff
@@ -79,54 +79,92 @@ internal static void Patch()
 
         public static bool NumCellsInRadius(out int __result, float radius)
         {
-            if (radius < 1f) // special case since we start scanning from the *previous* index, which would be negative in this instance.
+            // Special cases
+            if (radius < 2f || radius >= MAX_RADIUS)
             {
-                __result = 1;
-                return false;
-            }
-            var radialPatternNumCells = RadialPatternNumCells; // cache friendly local references
-            var radialPatternRadii = RadialPatternRadii;
-            if (radius >= MAX_RADIUS)
-            {
-                if (radius > MAX_RADIUS)
+                if (radius < 0f)
+                    __result = 0;
+                else if (radius < 1f)
+                    __result = 1;
+                else if (radius < 1.414213562f)
+                    __result = 5;
+                else if (radius < 2f)
+                    __result = 9;
+                else
                 {
-                    Log.Error($"Requested radius {radius} is beyond max. Truncating to {Harmony_GenRadial.MAX_RADIUS}.");
+                    __result = RadialPatternCount;
+                    CheckRadius(radius);
                 }
-                __result = RadialPatternCount;
                 return false;
             }
-            float radsq = radius * radius;
-            int r = (int)radsq;
-            int count = radialPatternNumCells[r - 1];
-            /*
-              If we raise the max radius above about 200, binary search becomes faster.
-              Below 200, the match will reliably be found within 64 tries, where linear memory access dominates.
-            int max_count = RadialPatternNumCells[r];
-            int span;
-            */
+
+            int radiusSquaredInt = (int)(radius * radius);
+            // The answer is at least (25/8 * radius * radius - 11)
+            int lowerBound = ((radiusSquaredInt * 25) >> 3) - 11;
```

Comment:
It's generally not a good idea in modern languages to try to be smarter than the compiler. You mean `/ 8`, and the compiler will already be smart enough to use a bitshift to do that if it really is the best way (and if it *isn't*, it will probably be smart enough to replace it with whatever register magic *is*). So all `>>3` does is make it harder for the humans reading it to follow.